add utility to compute self time of events

davidchencsl opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #79631

Purpose:
Compute self time for the experimental_event_tree
Test Plan:
Added test_utils_compute_self_time in test_profiler.py that verifies the computation.

davidchencsl committed:
add utility to compute self time of events

[ghstack-poisoned]

davidchencsl committed:
Update on "add utility to compute self time of events"


Purpose:
Compute self time for the experimental_event_tree
Test Plan:
Added test_utils_compute_self_time in test_profiler.py that verifies the computation.

[ghstack-poisoned]

robieta commented on torch/profiler/utils.py:
```diff
@@ -0,0 +1,49 @@
+from collections import deque
```
We should name this `_utils.py` to avoid introducing symbols to the public namespace.

davidchencsl replied:
Resolved

robieta reviewed (approved):
I had a single minor nit, but overall looks great!

davidchencsl committed:
Update on "add utility to compute self time of events"


Purpose:
Compute self time for the experimental_event_tree
Test Plan:
Added test_utils_compute_self_time in test_profiler.py that verifies the computation.

[ghstack-poisoned]

davidchencsl committed:
Update on "add utility to compute self time of events"


Purpose:
Compute self time for the experimental_event_tree
Test Plan:
Added test_utils_compute_self_time in test_profiler.py that verifies the computation.

[ghstack-poisoned]

davidchencsl committed:
Update on "add utility to compute self time of events"


Purpose:
Compute self time for the experimental_event_tree
Test Plan:
Added test_utils_compute_self_time in test_profiler.py that verifies the computation.

[ghstack-poisoned]

davidchencsl commented:
@pytorchmergebot merge -g
